sdk/core/System.ClientModel/tests/gen.unit/ContextGeneratorTests.cs

```diff
@@ -1140,5 +1140,456 @@ public void Write(Utf8JsonWriter writer, ModelReaderWriterOptions options) { }
+            StringAssert.Contains("#pragma warning restore CS0618", otherModelText);
+
+            Assert.AreEqual(2, result.GenerationSpec!.TypeBuilders.Count);
+            Assert.AreEqual("TEST001", result.GenerationSpec.TypeBuilders[0].Type.ExperimentalDiagnosticId);
+            Assert.IsNull(result.GenerationSpec.TypeBuilders[1].Type.ExperimentalDiagnosticId);
+
+            Assert.AreEqual(ObsoleteLevel.Warning, result.GenerationSpec.TypeBuilders[1].Type.ObsoleteLevel);
+        }
+
+        [Test]
+        public void DepWithExperimentalModel()
+        {
+            string depSource =
+$$"""
+  using System;
+  using System.ClientModel.Primitives;
+  using System.Diagnostics.CodeAnalysis;
+  using System.Text.Json;
+
+  namespace TestDependency
+  {
+  #pragma warning disable TEST001 // Experimental type
+      [ModelReaderWriterBuildable(typeof(JsonModel))]
+  #pragma warning restore TEST001 // Experimental type
+      public partial class LocalContext : ModelReaderWriterContext { }
+
+      [Experimental("TEST001")]
+      public class JsonModel : IJsonModel<JsonModel>
+      {
+          JsonModel IJsonModel<JsonModel>.Create(ref System.Text.Json.Utf8JsonReader reader, ModelReaderWriterOptions options) => new JsonModel();
+          JsonModel IPersistableModel<JsonModel>.Create(BinaryData data, ModelReaderWriterOptions options) => new JsonModel();
+          string IPersistableModel<JsonModel>.GetFormatFromOptions(ModelReaderWriterOptions options) => "J";
+          void IJsonModel<JsonModel>.Write(System.Text.Json.Utf8JsonWriter writer, ModelReaderWriterOptions options) { }
+          BinaryData IPersistableModel<JsonModel>.Write(ModelReaderWriterOptions options) => BinaryData.Empty;
+      }
+  }
+  """;
+
+            Compilation depCompilation = CompilationHelper.CreateCompilation(depSource, assemblyName: "TestDependency");
+            // The dependency compilation should not have experimental warnings
+            var depResult = CompilationHelper.RunSourceGenerator(depCompilation, out var newDepCompilation);
+
+            Assert.IsNotNull(depResult.GenerationSpec);
+            Assert.AreEqual("LocalContext", depResult.GenerationSpec!.Type.Name);
+            Assert.AreEqual("TestDependency", depResult.GenerationSpec.Type.Namespace);
+            Assert.AreEqual(0, depResult.Diagnostics.Length);
+            Assert.AreEqual("public", depResult.GenerationSpec!.Modifier);
+            Assert.AreEqual(1, depResult.GenerationSpec.TypeBuilders.Count);
+
+            string source =
+$$"""
+  using System;
+  using System.ClientModel.Primitives;
+  using TestDependency;
+
+  namespace TestProject
+  {
+  #pragma warning disable TEST001 // Experimental type
+      [ModelReaderWriterBuildable(typeof(JsonModel))]
+  #pragma warning restore TEST001 // Experimental type
+      public partial class MyLocalContext : ModelReaderWriterContext { }
+
+      public class Caller
```

Comment:
nit: I know there is a pattern that this exists but its not needed because we no longer do invocation discovery.  Its redundant to have this in here so we can remove.

*Note* multiple instances of this but only marking this one.